### DISCUSSION TITLE
How to make MSBuild find VSGlobals.vsct during command line builds

### DISCUSSION THREAD

**Discussion by pallibj** [VsixCommunity/Community.VisualStudio.Toolkit#247] [Q&A] [answered]
Using the VS2022 Community Toolkit to extend VS.

Inside my .vsct file I do a few includes, where Include href="VSGlobals.vsct" is one of them.

Building my solution from within VS2022 works fine.

However, if I clean the project from all non source-controlled file and do a full rebuild using MSBuild command line, I hit an error where the build can't find the VSGlobals.vsct file. The error messages from MSBuild is simply "error VSCT1118: File not found: Unable to locate 'VSGlobals.vsct' on the include paths provided".  If I then once (re)build from within VS2022 and then do a full MSBuild, the MSBuild works fine.

I know that I can copy this file into my project and it will most likely build successfully using MSBuild, but I don't really want to do that unless I have to.

Any idea what I need to do in order to make the MSBuild process find this file on the first full rebuild?

Palli

**Comment by CZEMacLeod** [comment, marked as answer]
@pallibj Are you doing a nuget restore before the build?

**Reply by pallibj** [reply]
> @pallibj Are you doing a nuget restore before the build?

Good question !

For some reason the nuget.targets is picking up a peculiar version number for my project, which makes it invalid because it is prefixed by two non-numeric characters, which is not valid for nuget versions. I have no idea where it is picking up that version number, but I agree with you that lack of nuget restore before the build is the most likely cause for this problem. 

Thanks,
Palli

**Reply by pallibj** [reply]
> @pallibj Are you doing a nuget restore before the build?

Sometimes one gets lost and doesn't see the forest for the trees!

Of course this was a lack of nuget restore before build. 
Thanks for your to the point question which let me to resolution of  the problem.

Palli
